relay-server/src/actors/outcome.rs

```diff
@@ -569,8 +569,10 @@ impl Service for HttpOutcomeProducer {
         tokio::spawn(async move {
             loop {
                 tokio::select! {
-                    Some(message) = rx.recv() => self.handle_message(message),
+                    biased;
```

Comment:
nit: I would add an inline explanation here, something like

```suggestion
                    biased; // poll flushes before messages
```